serverless-message-processing/template.yaml

```diff
@@ -0,0 +1,175 @@
+Description: Serverless message processing architecture
+
+Resources:
+  ApiGateway:
+    Type: AWS::Serverless::Api
+    Properties:
+      StageName: prod
+      DefinitionBody:
+        swagger: '2.0'
+        info:
+          title: SQS API
+          version: '1.0'
+        schemes:
+          - https
+        paths:
+          /message:
+            post:
+              consumes:
+                - application/json
+              produces:
+                - application/json
+              responses:
+                '200':
+                  description: OK
+                  schema:
+                    type: object
+              x-amazon-apigateway-integration:
+                credentials: !GetAtt ApiGatewayRole.Arn
+                type: aws
+                uri: !Sub "arn:aws:apigateway:${AWS::Region}:sqs:path/${AWS::AccountId}/${MainQueue.QueueName}"
+                httpMethod: POST
+                responses:
+                  default:
+                    statusCode: "200"
+                requestParameters:
+                  integration.request.header.Content-Type: "'application/x-www-form-urlencoded'"
+                requestTemplates:
+                  application/json: |
+                    Action=SendMessage&MessageBody=$input.body
+                passthroughBehavior: never
+                integrationResponses:
+                  '200':
+                    statusCode: '200'
+                    responseTemplates:
+                      application/json: |
+                        {
+                          "message": "Message sent successfully",
+                          "messageId": "$input.path('$.SendMessageResponse.SendMessageResult.MessageId')"
+                        }
+
+  ApiGatewayRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: apigateway.amazonaws.com
+            Action: sts:AssumeRole
+      Policies:
+        - PolicyName: ApiGatewaySQSPolicy
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action: sqs:SendMessage
+                Resource: !GetAtt MainQueue.Arn
+
+  MainQueue:
+    Type: AWS::SQS::Queue
+    Properties:
+      RedrivePolicy:
+        deadLetterTargetArn: !GetAtt AutomatedDLQ.Arn
+        maxReceiveCount: 3
+      VisibilityTimeout: 30
+      MessageRetentionPeriod: 345600  
+      ReceiveMessageWaitTimeSeconds: 20 
+
+  AutomatedDLQ:
+    Type: AWS::SQS::Queue
+    Properties:
+      MessageRetentionPeriod: 345600 
+
+
+  FatalDLQ:
+    Type: AWS::SQS::Queue
+    Properties:
+      MessageRetentionPeriod: 1209600 
+
+  ProcessMessages:
+    Type: AWS::Serverless::Function
+    Properties:
+      Handler: app.lambda_handler
+      Runtime: python3.9
+      CodeUri: ./functions/processor/
+      MemorySize: 128
+      Timeout: 29  
```

Comment:
Is this always going to be adequate as this scales out? Consider parameterising this value (for both functions) and including more comments about the when it's declared at the top of this YAML doc describing when people might want to increase it.